.gitignore

```diff
@@ -23,3 +23,4 @@
 /.pydevproject
 /.idea
 /*.iml
+*.swp
```

Comment:
Long story, short: in this particular case, it's probably okay to just do a global ignore at the root of the project like you did, but it's also probably better to ensure that it's only those that are dotfiles. In other words, use `.*.swp` instead of `*.swp`, in case that suffix is used for other things we might care about later.